libnmstate/state.py

```diff
@@ -402,6 +403,13 @@ def merge_interfaces(self, other_state):
             other_iface_state = origin_other_state.interfaces[name]
             if iface_state.get(Interface.TYPE) == LinuxBridge.TYPE:
                 merge_linux_bridge_ports(iface_state, other_iface_state)
+            if iface_state.get(Interface.TYPE) == InterfaceType.BOND:
```

Comment:
I think this fits an `elif`.

I find this post-merge "fix" a bit odd.
What do you think about identifying that the bond  mode changed, then, remove the `options` subtree from the current before the `dict_update` is called.
With this, you can have two functions in the `applier.bond`, one for checking if the bond mode changed and another for initializing the `options` subtree. When initializing, we can add the logging.

Reply:
Fixed the `elif`.

I don't want to touch the `current_state` which could cause follow up functions got unexpected data.
We can reuse this function in my on-going big change on merge and metadata process.